ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/MainnetContractCreationProcessor.java

```diff
@@ -113,6 +114,9 @@ public void start(final MessageFrame frame) {
           "Contract creation error: account as already been created for address {}",
           frame.getContractAddress());
       frame.setState(MessageFrame.State.EXCEPTIONAL_HALT);
+    } else if (contract instanceof ReadOnlyMutableAccount) {
```

Comment:
I don't think we need this. Public contracts created from private contract calls would go through a child message frame

Reply:
I've run code coverage and this seems to be right. Will remove.